c++/src/capnp/serialize.c++

```diff
@@ -301,6 +305,9 @@ void writeMessage(kj::OutputStream& output, kj::ArrayPtr<const kj::ArrayPtr<cons
 StreamFdMessageReader::~StreamFdMessageReader() noexcept(false) {}
 
 void writeMessageToFd(int fd, kj::ArrayPtr<const kj::ArrayPtr<const word>> segments) {
+#ifdef _WIN32
+  KJ_REQUIRE(_setmode(fd, _O_BINARY) == _O_BINARY, "Tried to write a message to a file descriptor that is in text mode. Set the file descriptor to binary mode by calling the _setmode Windows CRT function.");
```

Comment:
I think we should probably set the FD back to text mode before throwing.

```suggestion
    auto oldMode = _setmode(fd, _O_BINARY);
    if (oldMode != _O_BINARY) {
      _setmode(fd, oldMode);
      KJ_FAIL_REQUIRE("Tried to write a message to a file descriptor that is in text mode. Set the "
          "file descriptor to binary mode by calling the _setmode Windows CRT function, or passing "
          "_O_BINARY to _open().");
    }
```

(Also added a mention of _open() to the error message, and applied line wrapping to comply with KJ style rules.)